homeassistant/components/xiaomi_aqara/binary_sensor.py

```diff
@@ -476,19 +476,35 @@ def parse_data(self, data, raw_data):
             self._last_action = data[self._data_key]
 
         if 'rotate' in data:
-            self._hass.bus.fire('xiaomi_aqara.cube_action', {
-                'entity_id': self.entity_id,
-                'action_type': 'rotate',
-                'action_value': float(data['rotate'].replace(",", "."))
-            })
+            if isinstance(data['rotate'], int):
```

Comment:
You can't just copy paste the whole code. Instead, process the rotate information before firing the event.